### PR TITLE
Implement locale

### PR DESCRIPTION
**Pull request by NeumimTo**
# Description
Implements mockplayer.getLocale/locale
Adds setLocale method to update its value

# Checklist
The following items should be checked before the pull request can be merged.
- [ x] Code follows existing style.
- [ ] Unit tests added (if applicable).

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 1]
```diff
@@ -2983,4 +2983,8 @@ public void sendMessage(@NotNull ChatMessageType position, @NotNull BaseComponen
 
 	}
 
+	public void setLocale(@NotNull Locale locale) {
+		Objects.requireNonNull(locale, "Player`s Locale must be not null");
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java]: Please use Preconditions for null checks.

**Review by Insprill — CHANGES_REQUESTED** [review]
Does the locale setting effect other functionality that should be implemented?

I'm also not sure what you did with your branch, but can you rebase on the latest master to fix the duplicate commits?

**Comment by NeumimTo** [comment]
>Does the locale setting effect other functionality that should be implemented?

Im not aware of any. This is just a simple implementation for plugins that send messages to clients depending on their locale

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 2]
```diff
@@ -2983,4 +2983,8 @@ public void sendMessage(@NotNull ChatMessageType position, @NotNull BaseComponen
 
 	}
 
+	public void setLocale(@NotNull Locale locale) {
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:2991]: Please add a JavaDoc to this.

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 3]
```diff
@@ -2983,4 +2983,8 @@ public void sendMessage(@NotNull ChatMessageType position, @NotNull BaseComponen
...
+		Preconditions.checkNotNull(locale, "Player`s Locale must be not null");
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java]:
```suggestion
		Preconditions.checkNotNull(locale, "locale cannot be null");

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 4]
```diff
@@ -2477,8 +2478,7 @@ public int getClientViewDistance()
 	@Override
 	public @NotNull Locale locale()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return locale;
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java]:
```suggestion
		return this.locale;

**Review by Insprill — CHANGES_REQUESTED** [review]
This still needs to be properly rebased.

**Comment by NeumimTo** [comment]
commits should be fixed now

**Review by Insprill — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**